perfkitbenchmarker/benchmarks/hadoop_terasort_benchmark.py

```diff
@@ -36,21 +36,23 @@
 FLAGS = flags.FLAGS
 
 BENCHMARK_INFO = {'name': 'hadoop_terasort',
-                  'description': 'Runs Terasort',
+                  'description': 'Runs Terasort. Control the number of VMs '
+                  'with --num_vms.',
                   'scratch_disk': True,
-                  'num_machines': 9}
+                  'num_machines': None}
+DEFAULT_VM_COUNT = 9
 
-HADOOP_VERSION = '2.5.2'
-HADOOP_URL = ('http://apache.mirrors.tds.net/hadoop/common/hadoop-%s/'
-              'hadoop-%s.tar.gz') % (HADOOP_VERSION, HADOOP_VERSION)
-
-DATA_FILES = ['core-site.xml.j2', 'yarn-site.xml.j2', 'hdfs-site.xml',
-              'mapred-site.xml', 'hadoop-env.sh', 'slaves.j2']
 NUM_BYTES_PER_ROW = 100
+NUM_MB_PER_ROW = NUM_BYTES_PER_ROW / (1024.0 ** 2)
 
 
 def GetInfo():
-  return BENCHMARK_INFO
+  info = BENCHMARK_INFO.copy()
+  if FLAGS.num_vms == 1:  # Unchanged from default
+    info['num_machines'] = 9
```

Comment:
I think you can test FLAGS['num_vms'].present . Dot syntax gets the flag
value, [] syntax gets the Flag object itself.

On Wed, May 6, 2015 at 10:55 AM, Connor McCoy <notifications@github.com>
wrote:

> In perfkitbenchmarker/benchmarks/hadoop_terasort_benchmark.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/249#discussion_r29783904>
> :
>
> >
> >
> >  def GetInfo():
> > -  return BENCHMARK_INFO
> > +  info = BENCHMARK_INFO.copy()
> > +  if FLAGS.num_vms == 1:  # Unchanged from default
> > +    info['num_machines'] = 9
>
> I modified this to be a bit clearer. The benchmark can't run on a single
> machine, but I would like to keep the existing behavior of running without
> additional arguments from the user. The help text will show 9 VMs as the
> default.
>
> Agreed that checking num_vms is a bit hacky, but I don't know of a way of
> checking if a flag was set on the command line.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/249/files#r29783904>
> .
>